test/dynamo/test_repros.py

```diff
@@ -2700,6 +2725,89 @@ def f(x, y):
             gm(torch.zeros(6, 4), torch.tensor(2)),
         )
 
+    def test_hf_xsoftmax_inference(self):
```

Comment:
I'm not familiar with the dynamo code, but to check, does this mean after this PR:
- if the autograd.Function is passed inputs that do not require_grad, then dynamo traces through the forward pass
- if the autograd.Function is passed inputs that do require_grad, then dynamo graph breaks on the autograd.Function ?

Reply:
> if the autograd.Function is passed inputs that do not require_grad, then dynamo traces through the forward pass

Yes correct, though this is the same as the old behavior.  Just now it is more likely to succeed since we support the `ctx` object better.

> if the autograd.Function is passed inputs that do require_grad, then dynamo graph breaks on the entire autograd.Function ?

Yes, but then dynamo will pick up and compile the autograd.Function's frame.  So you end up with 3 graphs (before, after, and the autograd.Function).